File: DailyChallenges/CPP/LeetCode_1491.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        int lmin = INT_MAX, lmax = INT_MIN;
        long long sum = 0;
        for (int s:salary){
            lmin = min(s, lmin);
            lmax = max(s, lmax);
            sum += s;
        }
        return (double)(sum - lmin - lmax) / (salary.size()-2);
    }
};

File: DailyChallenges/Scala/LeetCode_1491.scala
object Solution {
    def average(salary: Array[Int]): Double = {
        val r = salary.foldLeft((Int.MinValue, Int.MaxValue, 0L)){(t, s) => (t._1 max s, t._2 min s, t._3 + s)}
        (r._3 - r._1 - r._2).toDouble / (salary.length - 2) 
    }
}
